components/layout/inline.rs

```diff
@@ -350,7 +389,7 @@ impl LineBreaker {
     fn flush_current_line(&mut self) {
         debug!("LineBreaker: flushing line {}: {:?}", self.lines.len(), self.pending_line);
         self.strip_trailing_whitespace_from_pending_line_if_necessary();
-        self.lines.push(self.pending_line);
+        self.lines.push(self.pending_line.clone());
```

Comment:
I'm a bit worried about the performance implications of doing this. It'd probably be better to have `reset_line()` swap out the current line and return the old one. What do you think?